src/ui/public/vislib/lib/axis_labels.js

```diff
@@ -88,28 +84,24 @@ export default function AxisLabelsFactory(Private) {
     filterAxisLabels() {
       const self = this;
       const config = this.config;
-      let startX = 0;
-      let maxW;
-      let par;
-      let myX;
-      let myWidth;
-      let halfWidth;
+      let startPos = 0;
       let padding = 1.1;
 
       return function (selection) {
         if (!config.get('labels.filter')) return;
 
         selection.selectAll('.tick text')
         .text(function (d) {
-          par = d3.select(this.parentNode).node();
-          myX = self.axisScale.scale(d);
-          myWidth = par.getBBox().width * padding;
-          halfWidth = myWidth / 2;
-          maxW = $(config.get('rootEl')).find(config.get('elSelector')).width();
+          const par = d3.select(this.parentNode).node();
+          const el = $(config.get('rootEl')).find(config.get('elSelector'));
+          const maxSize = config.isHorizontal() ? el.width() : el.height();
+          const myPos = config.isHorizontal() ? self.axisScale.scale(d) : maxSize - self.axisScale.scale(d);
+          const mySize = (config.isHorizontal() ? par.getBBox().width : par.getBBox().height) * padding;
+          const halfSize = mySize / 2;
 
-          if ((startX + halfWidth) < myX && maxW > (myX + halfWidth)) {
-            startX = myX + halfWidth;
-            return config.get('labels.axisFormatter')(d);
+          if ((startPos + halfSize) < myPos && maxSize > (myPos + halfSize)) {
+            startPos = myPos + halfSize;
+            return this.innerHTML;
```

Comment:
we are not setting it, just reading it. this part filters labels if there are to many, in case label should not be filtered it returns it AS IS ... so if your label would happen to include some html it would also work.